pkg/kvstore/etcd_test.go

```diff
@@ -1642,3 +1642,23 @@ func TestGetSvcNamespace(t *testing.T) {
 		})
 	}
 }
+func TestShuffleEndpoints(t *testing.T) {
+	s1 := []string{"1", "2", "3"}
+	s2 := make([]string, len(s1))
+	copy(s2, s1)
+	shuffleEndpoints(s2)
+
+	if len(s1) != len(s2) {
+		t.Errorf("Shuffle() resulted in slices of inequal length")
+	}
+
+	same := 0
+	for i := range s1 {
+		if s1[i] == s2[i] {
+			same++
+		}
+	}
+	if same == len(s1) {
+		t.Errorf("Shuffle() did not modify s2")
```

Comment:
Your array slice has only three elements, so the number of possible permutations is very low. Unless I'm missing something, this test function will have `s1 == s2` (and make the test fail) once in every six runs on average. You might want a longer string and/or running this a couple of times in a loop to fix that?

Reply:
Yes, I was able to reproduce. I actually had to use more than 3 entries as with 3 entries, `Shuffle()` generated the same sequence even with differing seed. Given that 3 is a very common number of etcd endpoints, I'm looking into a better way to Shuffle.